File: chad/src/main/java/org/woahoverflow/chad/commands/info/GuildInfo.java
package org.woahoverflow.chad.commands.info;

import org.woahoverflow.chad.framework.Command;
import org.woahoverflow.chad.framework.handle.MessageHandler;
import sx.blah.discord.handle.impl.events.guild.channel.message.MessageReceivedEvent;
import sx.blah.discord.handle.obj.IUser;
import sx.blah.discord.util.EmbedBuilder;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.HashMap;
import java.util.List;

public class GuildInfo implements Command.Class{
    @Override
    public final Runnable run(MessageReceivedEvent e, List<String> args) {
        return () -> {
            // Creates an embed builder and applies the title
            EmbedBuilder embedBuilder = new EmbedBuilder();
            embedBuilder.withTitle("Guild : " + e.getGuild().getName());

            // Gets the human to bot.
            int human = 0;
            int bot = 0;
            for (IUser u : e.getGuild().getUsers())
            {
                if (!u.isBot()) {
                    human++;
                } else {
                    bot++;
                }
            }

            // Applies the description
            embedBuilder.withDesc(
                    "Owner `"+e.getGuild().getOwner().getName()+ '`' +
                            "\nRole Amount `"+e.getGuild().getRoles().size()+ '`' +
                              "\nHuman to Bots `"+human+ '/' +bot+ '`' +
                              "\nUser Amount `"+e.getGuild().getUsers().size()+ '`' +
                              "\nVoice Channels `"+e.getGuild().getVoiceChannels().size()+ '`' +
                              "\nText Channels `"+e.getGuild().getChannels().size()+ '`' +
                              "\nCategories `"+e.getGuild().getCategories().size()+ '`' +
                              "\nCreation Date `"+new SimpleDateFormat("MM/dd/yyyy").format(Date.from(e.getGuild().getCreationDate()))+ '`'
            );

            // Adds the guild's image and sends.
            embedBuilder.withImage(e.getGuild().getIconURL());
            new MessageHandler(e.getChannel()).sendEmbed(embedBuilder);
        };
    }

    @Override
    public final Runnable help(MessageReceivedEvent e) {
        HashMap<String, String> st = new HashMap<>();
        st.put("guildinfo", "Gets information about the guild.");
        return Command.helpCommand(st, "Guild Info", e);
    }
}

File: chad/src/main/java/org/woahoverflow/chad/commands/info/GuildInfo.kt
package org.woahoverflow.chad.commands.info

import org.woahoverflow.chad.framework.handle.MessageHandler
import org.woahoverflow.chad.framework.obj.Command
import sx.blah.discord.handle.impl.events.guild.channel.message.MessageEvent
import sx.blah.discord.util.EmbedBuilder
import java.text.SimpleDateFormat
import java.util.*

/**
 * Gets info about the current guild
 *
 * @author sho
 */
class GuildInfo : Command.Class {
    override fun run(e: MessageEvent, args: List<String>): Runnable {
        return Runnable {
            // Creates an embed builder and applies the title
            val embedBuilder = EmbedBuilder()

            // Gets the human to bot.
            var human = 0
            var bot = 0
            for (u in e.guild.users) {
                if (!u.isBot)
                    human++
                else
                    bot++
            }

            // Applies the description
            embedBuilder.withDesc(
                    "**Name** : `${e.guild.name}`" +
                            "\n**Owner** : `${e.guild.owner.name}`" +
                            "\n**Role Amount** : `${e.guild.roles.size}`" +
                            "\n**Human to Bots** : `$human/$bot`"+
                            "\n**User Amount** : `${e.guild.users.size}`" +
                            "\n**Voice Channels** : `${e.guild.voiceChannels.size}`" +
                            "\n**Text Channels** : `${e.guild.channels.size}`" +
                            "\n**Categories** : `${e.guild.categories.size}`" +
                            "\n**Creation Date** : `${SimpleDateFormat("MM/dd/yyyy").format(Date.from(e.guild.creationDate))}`"
            )

            // Adds the guild's image and sends.
            embedBuilder.withImage(e.guild.iconURL)
            MessageHandler(e.channel, e.author).sendEmbed(embedBuilder)
        }
    }

    override fun help(e: MessageEvent): Runnable {
        val st = HashMap<String, String>()
        st["guildinfo"] = "Gets information about the guild."
        return Command.helpCommand(st, "Guild Info", e)
    }
}
